mlir/lib/CAPI/Transforms/Rewrite.cpp

```diff
@@ -70,6 +70,18 @@ MlirBlock mlirRewriterBaseGetBlock(MlirRewriterBase rewriter) {
   return wrap(unwrap(rewriter)->getBlock());
 }
 
+MlirOperation
+mlirRewriterBaseGetOperationAfterInsertion(MlirRewriterBase rewriter) {
+  mlir::RewriterBase *base = unwrap(rewriter);
+  mlir::Block *block = base->getInsertionBlock();
+  auto it = base->getInsertionPoint();
+  if (it == block->end()) {
+    return {nullptr};
+  }
+
+  return wrap(std::addressof(*it));
```

Comment:
just curious - does this work without `std::addressof`?

Reply:
Ahh yup. Here I thought that `&*it` is not so clear and looks weird. So use `addressof` instead. But there's no semantic difference.

Reply:
sorry I goofed here `base->getInsertionPoint()` is not a `PyInsertionPoint` it's a `mlir::InsertionPoint`!

Reply:
(that's why we don't do `auto` usually 😉)

Reply:
Ahh it is actually an `mlir::Block::iterator`. 🤣

Reply:
Sorry for the unclear. I'm out right now so cannot change the type soon. I'll revisit it tonight.

Reply:
yea whatever it is it is (i just knew it wasn't `PyInsertionPoint`). anyway yea LLVM style guide says to avoid `auto` unless it's a very long template type.